source/Ship.cpp

```diff
@@ -3585,6 +3617,111 @@ void Ship::SetTargetSystem(const System *system)
 
 
 
+// Persistent targets for mission NPCs.
+const bool Ship::HasTravelDirective() const
+{
+	return !stopovers.empty() || destinationPlanet || destinationSystem;
+}
+
+
+
+const map<const Planet *, bool> Ship::GetStopovers() const
+{
+	return stopovers;
+}
+
+
+
+const bool Ship::AllStopoversVisited() const
+{
+	bool allVisited = true;
+	if(stopovers.empty())
+		return allVisited;
+
+	for(auto const &it : stopovers)
+		allVisited &= it.second;
+	return allVisited;
```

Comment:
we don't need the variable
```suggestion
	if(stopovers.empty())
		return true;

	for(const auto &it : stopovers)
		if(!it.second)
			return false;
	return true;
```